### PR TITLE
Restore `trainer.current_epoch` after tuning

### LINKED ISSUE #5587: Weird behavior when using Trainer.tune
I have two cells in a notebook in colab:

Cell1:
```python
litmodel = LitModel(num_classes, 10.00017, len(dstrain), 64, model)
trainer = Trainer(gpus=1, auto_lr_find=True, max_epochs=epochs)
trainer.tune(litmodel, train_dataloader=dltrain, val_dataloaders=dlval)
trainer.fit(litmodel, dltrain, dlval)
```

Cell2:
```python
litmodel = LitModel(num_classes, 10.00017, len(dstrain), 64, model)
trainer = Trainer(gpus=1, auto_lr_find=True, max_epochs=epochs)
trainer.tune(litmodel, train_dataloader=dltrain, val_dataloaders=dlval)
trainer2 = Trainer(gpus=1, auto_lr_find=True, max_epochs=epochs)
trainer2.fit(litmodel, dltrain, dlval)
```

Cell2 produces much better results. The loss right at the beginning of the first epoch is about 8 for Cell1 and 0.8 for Cell2. 

This is the configure_optimizers function I use:

```python
    def configure_optimizers(self):
        optimizer = torch.optim.Adam(self.parameters(), lr= self.learning_rate)
        scheduler = {
                'scheduler': torch.optim.lr_scheduler.OneCycleLR(optimizer,
                     max_lr = self.learning_rate, 
                     total_steps = 2 * self.stepsize),
                'interval': 'step',
                'frequency': 1
        }
        return [optimizer], [scheduler]
```

pytorch-lightning version is 1.1.4.

This problem does not appear in pytorch-lightning version is 1.1.2.

---

Edit by @akihironitta: Let's discuss the below issue in #101.
```
Furthermore I get the following warning when running Cell1 right at the beginning of  trainer.fit():

> /usr/local/lib/python3.6/dist-packages/torch/optim/lr_scheduler.py:136: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
```

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #5587.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]

File: tests/tuner/test_lr_finder.py [hunk 1]
```diff
@@ -77,7 +77,13 @@ def test_trainer_reset_correctly(tmpdir):
...
 
     changed_attributes = [
-        'callbacks', 'logger', 'max_steps', 'auto_lr_find', 'accumulate_grad_batches', 'checkpoint_callback'
+        'callbacks',
+        'logger',
+        'max_steps',
+        'auto_lr_find',
+        'accumulate_grad_batches',
+        'checkpoint_callback',
+        'current_epoch',
```
Comment (awaelchli) [inline tests/tuner/test_lr_finder.py:86]: one could consider alphabetical ordering 😃

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]
